lib/std/Progress.zig

```diff
@@ -164,11 +197,64 @@ pub fn start(self: *Progress, name: []const u8, estimated_total_items: usize) *N
     };
     self.columns_written = 0;
     self.prev_refresh_timestamp = 0;
-    self.timer = std.time.Timer.start() catch null;
+    self.timer = time.Timer.start() catch null;
     self.done = false;
     return &self.root;
 }
 
+fn getTerminalWidth(self: Progress, file_handle: os.fd_t) !u16 {
+    if (builtin.os.tag == .windows) {
+        std.debug.assert(self.is_windows_terminal);
+        var info: windows.CONSOLE_SCREEN_BUFFER_INFO = undefined;
+        if (windows.kernel32.GetConsoleScreenBufferInfo(file_handle, &info) != windows.TRUE)
+            unreachable;
+        return @intCast(u16, info.dwSize.X);
+    } else if (builtin.os.tag == .linux) {
+        // TODO: figure out how to get this working on FreeBSD, macOS etc. too.
+        //       they too should have capabilities to figure out the cursor column.
+        var winsize: os.linux.winsize = undefined;
+        switch (os.errno(os.linux.ioctl(file_handle, os.linux.T.IOCGWINSZ, @ptrToInt(&winsize)))) {
+            .SUCCESS => return winsize.ws_col,
+            else => return error.Unexpected,
+        }
+    } else {
+        return error.Unsupported;
+    }
+}
+
+fn getTerminalCursorColumn(self: Progress, file: std.fs.File) !u16 {
+    // TODO: figure out how to get this working on FreeBSD, macOS etc. too.
+    //       they too should have termios or capabilities to figure out the terminal width.
+    if (builtin.os.tag == .linux and self.supports_ansi_escape_codes) {
+        // First, disable echo and enable non-canonical mode
+        // (so that no enter press required for us to read the output of the escape sequence below)
+        const original_termios = try os.tcgetattr(file.handle);
+        var new_termios = original_termios;
+        new_termios.lflag &= ~(os.linux.ECHO | os.linux.ICANON);
+        try os.tcsetattr(file.handle, .NOW, new_termios);
+        defer os.tcsetattr(file.handle, .NOW, original_termios) catch {
+            // Sorry for ruining your terminal
+        };
+
+        try file.writeAll("\x1b[6n");
+        var buf: ["\x1b[256;256R".len]u8 = undefined;
```

Comment:
These escapes are *not* portable (nor ascii as the doc-comment of the file suggests).
Getting these is why you need terminfo (rebutting https://github.com/ziglang/zig/issues/2931#issuecomment-1181882214)

Reply:
It seems to be widely supported: https://terminalguide.namepad.de/seq/csi_sn-6/
Can you elaborate on how this escape sequence is not portable for terminal emulators for Linux?

Reply:
> Terminals are more than just linux.

That's what the TODO comment above is for. This PR implements focuses on Linux and Windows.

> the terminal might be connecting over ssh/telnet/serial and you need to know the escapes for some other system

Sounds like a pretty advanced use case. This applies to a lot of cases (like simply invoking some `zig` command that causes colored output and such) and solving that seems to be out of scope of this PR. If you think this is an issue, you can open an issue for it but I'm guessing the majority of users won't face such issues.